modules/solid_mechanics/src/timeintegrators/CentralDifferenceDirect.C

```diff
@@ -0,0 +1,194 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "Assembly.h"
+#include "CentralDifference.h"
+#include "CentralDifferenceDirect.h"
+#include "MooseTypes.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+// libMesh includes
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/sparse_matrix.h"
+#include "DirichletBCBase.h"
+
+registerMooseObject("MooseApp", CentralDifferenceDirect);
+
+InputParameters
+CentralDifferenceDirect::validParams()
+{
+  InputParameters params = ExplicitTimeIntegrator::validParams();
+
+  params.addClassDescription(
+      "Implementation of Explicit/Forward Euler without invoking any of the nonlinear solver");
+
+  params.addParam<bool>("use_constant_mass",
+                        false,
+                        "If set to true, will only compute the mass matrix in the first time step, "
+                        "and keep using it throughout the simulation.");
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  return params;
+}
+
+CentralDifferenceDirect::CentralDifferenceDirect(const InputParameters & parameters)
+  : ExplicitTimeIntegrator(parameters),
+    _constant_mass(getParam<bool>("use_constant_mass")),
+    _mass_matrix(getParam<TagName>("mass_matrix_tag"))
+{
+  _fe_problem.setUDotOldRequested(true);
+  _fe_problem.setUDotDotRequested(true);
+  _fe_problem.setUDotDotOldRequested(true);
+}
+
+void
+CentralDifferenceDirect::computeTimeDerivatives()
+{
+  /*
+  Because this is called in NonLinearSystemBase
+  this should not actually compute the time derivatives.
+  Calculating time derivatives here will cause issues for the
+  solution update.
+  */
+  return;
+}
```

Comment:
Where can I check to see if the second time derivative is implemented correctly according to your math?

Reply:
nvmd, found it below.